File: py_topic_pkg/scripts/laser_scan_sub.py
#!/usr/bin/env python3

import rospy
from sensor_msgs.msg import LaserScan

class LaserSubNode:

    def __init__(self):
        # Publisher requires 3 paramters
        #  1. topic name
        #  2. topic msg type
        #  3. sub callback method
        self.laser_sub_ = rospy.Subscriber("scan", LaserScan, self.laser_cb)

    # first param of callback method is always topic msg
    def laser_cb(self, data):
        rospy.loginfo( len(data.ranges))

        print(f"""
        data.ranges[0]: {data.ranges[0]}
        data.ranges[90]: {data.ranges[90]}
        data.ranges[179]: {data.ranges[179]}
        data.ranges[270]: {data.ranges[270]}
        data.ranges[360]: {data.ranges[360]}
        """)

def laser_sub_node():
    rospy.init_node('laser_sub_node', anonymous=True)
    laser_sub_node = LaserSubNode()
    rospy.spin()

if __name__ == '__main__':
    try:
        laser_sub_node()
    except rospy.ROSInterruptException:
        pass

File: cpp_topic_pkg/src/laser_scan_sub.cpp
#include <ros/ros.h>
#include <geometry_msgs/Twist.h>
#include <sensor_msgs/LaserScan.h>

class LaserSubNode
{
private:
    ros::Subscriber laser_sub_;
    ros::Timer timer_;

public:
    LaserSubNode(ros::NodeHandle *nh) {
        ROS_INFO("Publisher and Subscriber initialized");
        laser_sub_ = nh->subscribe("scan", 10, &LaserSubNode::laserSubCallback, this);
    }

    void laserSubCallback(const sensor_msgs::LaserScan &data){
        
        ROS_INFO_STREAM("data.ranges[0]: " << data.ranges[0]);
        ROS_INFO_STREAM("data.ranges[90]: " << data.ranges[90]);
        ROS_INFO_STREAM("data.ranges[179]: " << data.ranges[179]);
        ROS_INFO_STREAM("data.ranges[270]: " << data.ranges[270]);
        ROS_INFO_STREAM("data.ranges[360]: " << data.ranges[360]);
        // std::cout << std::to_string(data.ranges[0]) << std::endl;

        ROS_INFO("Publisher and Subscriber initialized");

    }
};

int main(int argv, char** argc) {
    
    ros::init(argv, argc, "laser_sub_node");
    // ros::NodeHandle nh("my_namespace");
    ros::NodeHandle nh;
    
    LaserSubNode laser_sub_node(&nh);
    
    ros::spin();

    return 0;
}